pkg/aws/ec2_helpers.go

```diff
@@ -67,12 +68,25 @@ func (ec AWSEc2DynamicConfig) createClient(kubeClient client.Client, ctx context
 }
 
 // configureInstance creates and returns an EC2 instance configuration.
-func (ec AWSEc2DynamicConfig) configureInstance(taskRunName string, instanceTag string, additionalInstanceTags map[string]string) *ec2.RunInstancesInput {
+func (ec AWSEc2DynamicConfig) configureInstance(taskRunName string, instanceTag string, additionalInstanceTags map[string]string) (*ec2.RunInstancesInput, error) {
+	// Validate that MacOS-specific fields are either all set or all empty
+	tenancySet := ec.Tenancy != ""
+	hostResourceGroupSet := ec.HostResourceGroupArn != ""
+	licenseConfigSet := ec.LicenseConfigurationArn != ""
+
+	if tenancySet || hostResourceGroupSet || licenseConfigSet {
+		if !tenancySet || !hostResourceGroupSet || !licenseConfigSet {
+			return nil, errors.New("MacOS dedicated host configuration requires all three fields: Tenancy, HostResourceGroupArn, and LicenseConfigurationArn must be set together")
+		}
+	}
+
```

Comment:
we don't usually validate this configuration, we rely on AWS doing this check. Let's follow this pattern also for this new fields.

```suggestion
func (ec AWSEc2DynamicConfig) configureInstance(taskRunName string, instanceTag string, additionalInstanceTags map[string]string) ec2.RunInstancesInput {

```